doc/api/permissions.md

```diff
@@ -595,6 +596,10 @@ There are constraints you need to know before using this system:
 
 #### Limitations and Known Issues
 
+* The permission model provides no security guarantees in the presence of
```

Comment:
I'd for sure note state that. I wrote https://github.com/nodejs/node/pull/54269 instead.

Reply:
What security guarantees does the permission model provide in the presence of malicious code?

Reply:
Which point of `permissions.md` states that the permission model guarantees protection against malicious code? It's also stated in our [Threat Model](https://github.com/nodejs/node/blob/main/SECURITY.md#the-nodejs-threat-model) that Node.js trusts in the code being run.

Reply:
> Which point of `permissions.md` states that the permission model guarantees protection against malicious code?

I don't think the user-facing API documentation says anything about this -- but it also does not deny it. From a user's perspective, it is very reasonable to assume that the permission model is supposed to protect against malicious code. After all, that was the motivation according to [the original post by James](https://www.nearform.com/insights/adding-a-permission-system-to-node-js/), and it is what Deno tells its users about its permission model -- and it is what most security feature are designed to do.

To me, as a user, it would be very surprising to see something advertised as a security mechanism if it only provides guarantees for trusted code. So if that's the case, then we should explain in the public documentation what exactly the goal of the permission model is. It can hardly be to protect against supply chain attacks if it does not protect against malicious code.

> It's also stated in our [Threat Model](https://github.com/nodejs/node/blob/main/SECURITY.md#the-nodejs-threat-model) that Node.js trusts in the code being run. If we add it here, we should add that mention to any Node.js API.

I agree that the threat model is not very visible to users. However, following that argument, if Node.js implicitly trusts all running code even with the permission model enabled, then we should also mention on the permissions page that the permission model does not protect against untrusted code (which includes malicious code).

Reply:
I understand your perspective, but the description under "Limitations & Known issues" doesn't align with my views. Adding such a strong statement would mean more than: "You should not run malicious code using Node.js", it sounds more: "Don't use Permission Model at all". This is not a limitation or something we have discovered after building this feature, due to how Node.js works and how this feature was developed, this is just expected. 

I agree that adding a note at the top of this file, similar to what we did for `node:vm` https://nodejs.org/api/vm.html, would not be that bad. I just want to make sure you understand my perspective that these statements could discourage users from using that feature, thereby affecting its development. It's important to note that this feature is experimental, and we could address most of these issues with a more efficient design.

Reply:
I agree that it puts a nice positive spin on the permission model :)